program_transformers/src/rollups/rollup_persister.rs

```diff
@@ -0,0 +1,616 @@
+                    if let Err(err) = self
+                        .download_rollup(&mut rollup_to_verify, &mut rollup)
+                        .await
+                    {
+                        error!("Error during rollup downloading: {}", err)
+                    };
+                }
+                &RollupPersistingState::SuccessfullyDownload => {
+                    if let Some(r) = &rollup {
+                        self.validate_rollup(&mut rollup_to_verify, r).await;
+                    } else {
+                        error!(
+                            "Trying to validate non downloaded rollup: {:#?}",
+                            &rollup_to_verify
+                        )
+                    }
+                }
+                &RollupPersistingState::SuccessfullyValidate => {
+                    if let Some(r) = &rollup {
+                        // TODO: Add retry?
+                        if let Err(e) = self.store_rollup_update(&mut rollup_to_verify, r).await {
+                            error!("Store rollup update: {}", e)
+                        };
+                    } else {
+                        error!(
+                            "Trying to store update for non downloaded rollup: {:#?}",
+                            &rollup_to_verify
+                        )
+                    }
+                }
+                &RollupPersistingState::FailedToPersist | &RollupPersistingState::StoredUpdate => {
+                    if let Err(e) = self.drop_rollup_from_queue(&rollup_to_verify).await {
+                        error!("failed to drop rollup from queue: {}", e);
+                    };
+                    info!(
+                        "Finish processing {} rollup file with {:?} state",
+                        &rollup_to_verify.url, &rollup_to_verify.rollup_persisting_state
+                    );
+                    statsd_histogram!(
+                        "rollup.persisting_latency",
+                        start_time.elapsed().as_millis() as u64
+                    );
+                    statsd_count!("rollup.total_processed", 1);
+                    return;
+                }
+            }
+        }
+    }
+
+    pub async fn get_rollup_to_verify(
+        &self,
+    ) -> Result<(Option<rollup_to_verify::Model>, Option<rollup::Model>), ProgramTransformerError>
+    {
+        let multi_txn = self.txn.begin().await?;
+        let condition = Condition::all()
+            .add(
+                rollup_to_verify::Column::RollupPersistingState
+                    .ne(RollupPersistingState::FailedToPersist),
+            )
+            .add(
+                rollup_to_verify::Column::RollupPersistingState
+                    .ne(RollupPersistingState::StoredUpdate),
+            )
+            .add(
+                rollup_to_verify::Column::RollupPersistingState
+                    .ne(RollupPersistingState::StartProcessing),
+            );
```

Comment:
If the state of the rollup is SuccessfullyDownload or SuccessfullyValidate then it will be picked up here by processor and continue processing from this state

Reply:
oh yeah I see.  Somehow thought it would get reset.